pkg/auth/service.go

```diff
@@ -44,6 +44,11 @@ type AuthorizationResponse struct {
 	Error   error
 }
 
+type NeededPermissions struct {
```

Comment:
rename, `MissingPermissions` or `PermissionsAudit` or `DenyHistory` w/e would be more appropriate?